builder/builder-next/exporter/wrapper.go

```diff
@@ -62,8 +78,26 @@ func (i *imageExporterInstanceWrapper) Export(ctx context.Context, src *exporter
 
 	desc := ref.Descriptor()
 	imageID := out[exptypes.ExporterImageDigestKey]
-	if i.callback != nil {
-		i.callback(ctx, imageID, desc)
+	if i.callbacks.Exported != nil {
+		i.callbacks.Exported(ctx, imageID, desc)
 	}
+
+	if i.callbacks.Named != nil {
+		for _, name := range strings.Split(out[string(exptypes.OptKeyName)], ",") {
+			ref, err := reference.ParseNormalizedNamed(name)
```

Comment:
I'm wondering if we're trying too hard here; my thinking is that `ExporterInstance.Export` would probably fail if any of the `OptKeyName` values was invalid. If it didn't fail then we _should _ assume that all of them were accepted.

From that perspective, we could just split the string, and call `i.callbacks.Named(ctx, namedTagged, desc)`

Reply:
Yeah it's unexpected that the parsing will fail at this point (as highlighted in comment). The intent here was to provide a `reference.Named` type instead of raw string to match what the value actually represents.